src/Application/Actions/Hooks/StripeUpdate.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MatchBot\Application\Actions\Hooks;
+
+use Doctrine\ORM\EntityManagerInterface;
+use MatchBot\Application\Actions\Action;
+use MatchBot\Application\Actions\ActionError;
+use MatchBot\Application\Actions\ActionPayload;
+use MatchBot\Domain\Donation;
+use MatchBot\Domain\DonationRepository;
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Serializer\SerializerInterface;
+use Stripe\Event as StripeEvent;
+
+/**
+ * @return Response
+ */
+class StripeUpdate extends Action
+{
+    private DonationRepository $donationRepository;
+    private EntityManagerInterface $entityManager;
+    private SerializerInterface $serializer;
+
+    public function __construct(
+        DonationRepository $donationRepository,
+        EntityManagerInterface $entityManager,
+        LoggerInterface $logger,
+        SerializerInterface $serializer
+    ) {
+        $this->donationRepository = $donationRepository;
+        $this->entityManager = $entityManager;
+        $this->serializer = $serializer;
+
+        parent::__construct($logger);
+    }
+
+    protected function action(): Response
+    {
+        $payload = $this->request->getBody();
+        $signature = $this->request->getHeaderLine('stripe-signature');
+        $webhookSecret = getenv('STRIPE_WEBHOOK_SIGNING_SECRET');
+
+        try {
+            $event = \Stripe\Webhook::constructEvent($payload, $signature, $webhookSecret);
+        } catch (\UnexpectedValueException $e) {
+            return $this->validationError("Invalid Payload", null);
+        } catch (\Stripe\Exception\SignatureVerificationException $e) {
+            return $this->validationError("Invalid Signature", null);
+        }
+
+        if ($event instanceof StripeEvent) {
+            /** @var Donation $donation */
+            $donation = $this->donationRepository->findOneBy(['transactionId' => $event->data->object->id]);
+
+            if ($donation) {
+                switch ($event->type) {
+                    case 'payment_intent.succeeded':
+                        $this->handlePaymentIntentSucceeded($event, $donation);
+                        break;
+                    default:
+                        return $this->validationError("Unsupported Action", null);
+                }
+            } else {
+                $logMessage = 'No Content';
+                $this->logger->warning($logMessage);
```

Comment:
I think this can just be an `info()` call, as we expect it in sandboxes when receiving data from others, and should probably include the event type / ID for reference.